vita3k/gui/src/live_area_dialog.cpp

```diff
@@ -0,0 +1,694 @@
+                const auto scal_str = size_text / 1.37f;
+                const auto scal_size_str = size_text * scal_str;
+                const auto str_size = 12.0f * scal_str;
+
+                auto size_str = ImVec2((str[host.io.title_id][frame.first][current_item[host.io.title_id][frame.first]].size() * 11.9f) * scal_str, str_size);
+
+                if (size_str.x > FRAME_SIZE.x)
+                    size_str.x = FRAME_SIZE.x - 32.0f;
+
+                auto str_pos_init = ImVec2((FRAME_SIZE.x - size_str.x) / 2.0f, (FRAME_SIZE.y - size_str.y) / 2.0f);
+
+                if (str_valign[host.io.title_id][frame.first] == "top")
+                    str_pos_init.y = 0.0f;
+                else if (str_valign[host.io.title_id][frame.first] == "bottom")
+                    str_pos_init.y = FRAME_SIZE.y - str_size;
+
+                const auto pos_str = ImVec2(pos_frame.x + (str_pos_init.x * scal.x) + (str_pos[host.io.title_id][frame.first].x * scal.x),
+                    pos_frame.y + (str_pos_init.y * scal.y) + (str_pos[host.io.title_id][frame.first].y * scal.y));
+
+                if (!gui.live_area.manual_dialog)
+                    ImGui::GetForegroundDrawList()->AddText(ImGui::GetFont(), ImGui::GetFontSize() * (scal_size_str * scal.x),
+                        pos_str, IM_COL32(R, G, B, 255), str[host.io.title_id][frame.first][current_item[host.io.title_id][frame.first]].c_str(),
+                        nullptr, size_frame.x - ((32.f * scal_str) * scal.x));
+            }
+        }
+    }
+
+    for (auto &flip : frame_autoflip[host.io.title_id])
+        if (flip.second != 0) {
+            if (last_time[host.io.title_id][flip.first] == 0)
+                last_time[host.io.title_id][flip.first] = current_time();
+
+            while (last_time[host.io.title_id][flip.first] + flip.second < current_time()) {
+                last_time[host.io.title_id][flip.first] += flip.second;
+
+                if (gui.live_items[host.io.title_id][flip.first].find(bg) != gui.live_items[host.io.title_id][flip.first].end()) {
+                    if (current_item[host.io.title_id][flip.first] != int(gui.live_items[host.io.title_id][flip.first][bg].size()) - 1)
+                        ++current_item[host.io.title_id][flip.first];
+                    else if (current_item[host.io.title_id][flip.first] == int(gui.live_items[host.io.title_id][flip.first][bg].size() - 1))
+                        current_item[host.io.title_id][flip.first] = 0;
+                } else if (gui.live_items[host.io.title_id][flip.first].find(img) != gui.live_items[host.io.title_id][flip.first].end()) {
+                    if (current_item[host.io.title_id][flip.first] != int(gui.live_items[host.io.title_id][flip.first][img].size()) - 1)
+                        ++current_item[host.io.title_id][flip.first];
+                    else if (current_item[host.io.title_id][flip.first] == int(gui.live_items[host.io.title_id][flip.first][img].size() - 1))
```

Comment:
```suggestion
                    else
```

same here, i'm pretty sure both statements are exact opposites